yjit/src/codegen.rs

```diff
@@ -2225,7 +2225,7 @@ fn gen_checktype(
             Opnd::mem(64, val, RUBY_OFFSET_RBASIC_FLAGS),
             Opnd::UImm(RUBY_T_MASK.into()));
         asm.cmp(object_type, Opnd::UImm(type_val.into()));
-        let ret_opnd = asm.csel_e(Opnd::UImm(Qfalse.into()), Opnd::UImm(Qtrue.into()));
+        let ret_opnd = asm.csel_ne(Opnd::UImm(Qfalse.into()), Opnd::UImm(Qtrue.into()));
```

Comment:
Now that checktype after send is properly tested, I needed to fix a couple of places in checktype to make tests pass. This line seems to be the same situation as https://github.com/Shopify/ruby/pull/352#issuecomment-1204544410.

Reply:
```suggestion
        let ret_opnd = asm.csel_e(Qtrue.into(), Qfalse.into());
```

I think it's more intuitive to read without the negation (just flip the order of the arguments), and we can also do `Qtrue.into()` and `Qfalse.into()` directly.

Reply:
agreed, applied your suggested changes https://github.com/Shopify/ruby/pull/357/commits/a633f5d46c1b271cc1e823ca092187ba0abd1eae